core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -29,31 +29,32 @@ class SparkException(
     cause: Throwable,
     errorClass: Option[String],
     errorSubClass: Option[String],
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
It is tested by at least QueryExecutionErrorsSuite "INVALID_BUCKET_FILE: error if there exists any malformed bucket files"